flow360/component/simulation/validation/validation_simulation_params.py

```diff
@@ -0,0 +1,37 @@
+"""
+validation for SimulationParams
+"""
+
+from flow360.component.flow360_params.flow360_fields import get_aliases
+from flow360.component.simulation.models.surface_models import Wall
+from flow360.component.simulation.outputs.outputs import SurfaceOutput
+
+
+def _check_consistency_wall_function_and_surface_output(v):
+    has_wall_function_model = False
+
+    models = v.models
+
+    if models is None:
+        return v
+
+    for model in models:
+        if isinstance(model, Wall) and model.use_wall_function:
+            has_wall_function_model = True
```

Comment:
maybe we can already return here?

Reply:
True! Added a `break` to avoid unnecessary loops. Implemented in `34d5dfe`